spec/abstractops.html

```diff
@@ -157,10 +219,25 @@ <h1>CreateDateTime ( …args )</h1>
     </p>
   </emu-clause>
 
-  <emu-clause id=sec-temporal-ToTimeLike aoid=ToTimeLike>
-    <h1>ToTimeLike ( _time_ )</h1>
-    <p>
-      <mark>TODO: Define.</mark>
-    </p>
+  <emu-clause id=sec-temporal-FormatSecondsStringPart aoid=FormatSecondsStringPart>
+    <h1>FormatSecondsStringPart ( _second_, _millisecond_, _microsecond_, _nanosecond_ )</h1>
+    <emu-alg>
+      1. If _second_, _millisecond_, _microsecond_, and _nanosecond_ are all 0, then
+        1. Return `""`.
+      1. Let _nanos_, _micros_, and _millis_ be `""`.
+      1. If _nanosecond_ is not 0, then
+        1. Set _nanos_ to _nanosecond_ formatted as a three-digit decimal number, padded to the left with zeroes if necessary.
```

Comment:
i believe these can use https://tc39.es/ecma262/#sec-stringpad instead of prose

Reply:
StringPad does a bunch of things that can throw exceptions, so I'd kinda prefer not to. Maybe we can introduce a separate abstract operation and also use it in the dozen or so places that ecma262 uses this wording.

Reply:
None of those exceptions are observable with primitive string inputs, with a length of 3 - I'm not sure why we'd want to make a different one.

Reply:
Will wait for a response on https://github.com/tc39/ecma262/issues/1737